crates/pallets/starknet/src/tests/utils.rs

```diff
@@ -112,3 +116,58 @@ pub fn build_get_balance_contract_call(account_address: StarkFelt) -> (EntryPoin
 
     (balance_of_selector, calldata)
 }
+
+pub fn create_resource_bounds() -> starknet_api::transaction::ResourceBoundsMapping {
+    let mut map = BTreeMap::new();
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
+        starknet_api::transaction::ResourceBounds { max_amount: 10000, max_price_per_unit: 12000 },
+    );
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
```

Comment:
You are inserting two times with key == L1Gas. You should probably use l2Gas for the second, no?